views/default/navigation/menu/elements/item.php

```diff
@@ -56,6 +56,11 @@
 	$item_vars['class'][] = "elgg-state-selected";
 }
 
+$badge = $item->badge;
```

Comment:
did you test this? maybe it needs to be $item->getData('badge') i am not sure... the tests are probably failing on this

Reply:
I am planning to do this..

```
if (is_numeric($item->badge)) {
$item->badge = format_number_function($item->badge);
}
```
I don't know why but phpunit test does not work on my localhost.

Reply:
`$item->badge` is working fine, I have tested the same in search feature.